packages/apollo/src/commands/client/__tests__/generate.test.ts

```diff
@@ -191,6 +191,28 @@ describe("client:codegen", () => {
       expect(fs.readFileSync("API.ts").toString()).toMatchSnapshot();
     });
 
+  test
```

Comment:
This hopefully will fix it! Note the removal of the `.` in the extension flag and the changes to the test to check for the right file

```suggestion
  test
    .fs({
      "schema.json": fullSchemaJsonString,
      "queryOne.graphql": simpleQuery.toString(),
      "my.config.js": defaultConfig
    })
    .command([
      "client:codegen",
      "--config=my.config.js",
      "--target=typescript",
      "--fileExtension=d.ts",
      "--outputFlat",
      "outDirectory"
    ])
    .it("writes types for typescript with a custom extension", () => {
      const files = fs.readdirSync("./outDirectory");
      const filePath = files.find(f => f === "SimpleQuery.d.ts");
      expect(filePath).toBeDefined();
      const file = fs.readFileSync(`./outDirectory/${filePath}`).toString();
      expect(file).toMatchSnapshot();
    });
```